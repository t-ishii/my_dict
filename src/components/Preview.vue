<template>
  <div class="preview" v-html="preview">
  </div>
</template>

<script>
import showdown from 'showdown'

export default {
  name: 'Preview',
  props: {
    text: ''
  },
  computed: {
    preview() {
      const converter = new showdown.Converter()
      return converter.makeHtml(this.text)
    }
  }
}
</script>
