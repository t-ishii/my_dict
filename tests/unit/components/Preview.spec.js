import { shallowMount } from '@vue/test-utils'
import Preview from '@/components/Preview.vue'

import showdown from 'showdown'

const toHtml = (markdown) => {
  const converter = new showdown.Converter()
  return converter.makeHtml(markdown)
}

describe('Preview.vue', () => {

  // 正しくパラメータがセットされるか
  it('Whether the parameter is set', () => {
    const propText = '# test'
    const wrapper = shallowMount(Preview, {
      propsData: {
        text: propText
      }
    })
    expect(wrapper.props().text).toBe(propText)
    expect(wrapper.html()).toMatchSnapshot()
  })

  // マークダウンが変換されているか
  it('Whether markdown has been converted', () => {
    const propText = '# test\n- test1\n- test2'
    const wrapper = shallowMount(Preview, {
      propsData: {
        text: propText
      }
    })
    expect(wrapper.html()).toBe(`<div class="preview">${toHtml(propText)}</div>`)
    expect(wrapper.html()).toMatchSnapshot()
  })

  // パラメータが渡されなくても動くか
  it('Whether it works even if no parameters are passed', () => {
    const wrapper = shallowMount(Preview)
    expect(wrapper.isVisible()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
