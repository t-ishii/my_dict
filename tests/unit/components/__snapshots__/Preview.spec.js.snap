// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview.vue Whether it works even if no parameters are passed 1`] = `<div class="preview"></div>`;

exports[`Preview.vue Whether markdown has been converted 1`] = `
<div class="preview">
  <h1 id="test">test</h1>
  <ul>
    <li>test1</li>
    <li>test2</li>
  </ul>
</div>
`;

exports[`Preview.vue Whether the parameter is set 1`] = `
<div class="preview">
  <h1 id="test">test</h1>
</div>
`;
